DNN Platform/Library/Services/Sitemap/CoreSitemapProvider.cs

```diff
@@ -65,7 +65,7 @@ public override List<SitemapUrl> GetUrls(int portalId, PortalSettings ps, string
                         (Null.IsNull(tab.StartDate) || tab.StartDate < DateTime.Now) &&
                         (Null.IsNull(tab.EndDate) || tab.EndDate > DateTime.Now) && this.IsTabPublic(tab.TabPermissions))
                     {
-                        if ((this.includeHiddenPages || tab.IsVisible) && tab.HasBeenPublished)
+                        if ((this.includeHiddenPages || tab.IsVisible) && tab.HasBeenPublished && Convert.ToBoolean(tab.TabSettings["AllowIndex"]))

```

Comment:
It looks like the behavior in other places is that `null` indicates `AllowIndex` is `true`, while using `Convert.ToBoolean` will convert it to `false`.  Can you copy the parsing logic from [`TabIndexer`](https://github.com/dnnsoftware/Dnn.Platform/blob/cc61549005ab5ea1c66fa91670931642a62893e4/DNN%20Platform/Library/Services/Search/TabIndexer.cs#L52-L55) (or, even better, introduce a computed property on `TabInfo` which encapsulates this logic from both places).

Also, you'll need to rebase this on the latest from `develop`, which includes your previous PR.  This PR will need to undo that other PR.  

Thanks!